plugins/rapid7_insightvm/komand_rapid7_insightvm/triggers/scan_completion/trigger.py

```diff
@@ -0,0 +1,245 @@
+
+            for scan in response.get("resources"):
+                if scan.get("status") == "finished":
+                    self.logger.info(f"Latest finished scan ID: {scan.get('id')}")
+                    return scan.get("id")
+
+
+class ScanQueries:
+    @staticmethod
+    def query_results_from_latest_scan(scan_id: int) -> str:
+        """
+        Generate an SQL query string needed to to retrieve all the necessary outputs
+
+        :param scan_id: Scan ID to query against
+        :return: The completed query string
+        """
+
+        return (
+            f"SELECT fasvi.scan_id, fasvi.asset_id, fasvi.vulnerability_id, dv.cvss_v3_score, dvr.source, daga.asset_group_id, dss.solution_id, dss.summary, dv.nexpose_id "  # nosec B608
+            f"FROM fact_asset_scan_vulnerability_instance AS fasvi "
+            f"JOIN dim_asset_group_asset AS daga ON (fasvi.asset_id = daga.asset_id) "
+            f"JOIN dim_vulnerability AS dv ON (fasvi.vulnerability_id = dv.vulnerability_id) "
+            f"JOIN dim_vulnerability_reference AS dvr ON (fasvi.vulnerability_id = dvr.vulnerability_id) "
+            f"JOIN dim_solution AS dss ON (dv.nexpose_id = dss.nexpose_id) "
+            f"WHERE fasvi.scan_id = {scan_id} "
+        )
+
+
+class Util:
+    @staticmethod
+    def filter_results(params: dict, csv_row: dict, results: dict) -> Union[None, dict]:
+        """
+        Filter the outputted results based on the user inputs.
+
+        :param params: Input params
+        :param csv_row: Dict row of the csv results
+        :param results: New object to append results to
+
+        :return: New object containing only the necessary fields for the required output.
+        """
+
+        # Input retrieval
+        asset_group = params.get(Input.ASSET_GROUP, None)
+        cve = params.get(Input.CVE, None)
+        source = params.get(Input.SOURCE, None)
+        cvss_score = params.get(Input.CVSS_SCORE, None)
+        severity = params.get(Input.SEVERITY, None)
+
+        # We retrieve this separately because we use it as a unique identifier for
+        # the filtering process
+        asset_id = int(csv_row.get("asset_id", 0))
+
+        new_dict = {
+            "asset_id": asset_id,
+            "vulnerability_info": [
+                {
+                    "vulnerability_id": csv_row.get("vulnerability_id", ""),
+                    "nexpose_id": csv_row.get("nexpose_id", ""),
+                    "cvss_v3_score": csv_row.get("cvss_v3_score", 0),
+                    "severity": csv_row.get("severity", ""),
+                    "solution_id": Util.strip_msft_id(csv_row.get("solution_id", "")),
+                    "solution_summary": csv_row.get("summary", ""),
+                }
+            ],
+        }
+
+        # If an input and it is not found, return None in place of the row to filter
+        # out the result
+        if asset_group and asset_group not in csv_row.get("asset_group_id", ""):
```

Comment:
I think returning None here will remove any old results? do we need to do `return {}` and in the loop:

```
for row in csv_report:
            results_dicts.update(Util.filter_results(params, row, results_dict))
```

Or we could do `return` to stop executing this method, if the loop was like this, as we'd be dynmically updating results in here, and don't have `return results` at the end?
```
for row in csv_report:
            Util.filter_results(params, row, results_dict)
```